### PR TITLE
[WIP] Add TSM app observability content

### LINKED ISSUE #826: Tanzu Service Mesh Identifying and Understanding Problems
Describe, with Tanzu Service Mesh,  how to Identify and understand where a problem is coming from (a service, between services, a cluster, a cloud, node-to-service correlation - GNS) and what are the symptoms and impact of the problem (e.g., latency, errors, etc.) and how long it has existed.

### PR DESCRIPTION
**Pull request by ahuffman**
Closes #826 
Signed-off-by: Andrew J. Huffman <ahuffman@vmware.com>

### CODE REVIEW

File: content/outcomes/application-observability/tsm-app-observability.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+date: '2021-07-13'
+lastmod: '2021-07-13'
+layout: single
+team:
+  - Mark Schweighardt
+  - Niran Even-Chen
+  - Andrew Huffman
+title: Tanzu Service Mesh Application Observability Introduction
```
Comment (zmb3) [inline content/outcomes/application-observability/tsm-app-observability.md:9]: Is this file really necessary? It seems the rest of the files have a direct line to a customer outcome, but this is more just introductory text. Can we remove this and add whatever intro is necessary in the other pages?

File: content/outcomes/application-observability/tsm-app-troubleshooting-day2.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+weight: 102
+---
+
+When a distributed application is experiencing issues it's not an easy task to figure out where to look. Tanzu service mesh provides views which can pinpoint faster where problems are coming from and reduce the time to identify the issue. One could start from the views we discussed in the day 1 section to see whether the application is experiencing issues and where the issues are, by looking at the "Golden metrics" and the dependencies between the different services. In addition, Tanzu service mesh provides more extensive views which can become helpful in the troubleshooting task. After identifying an area of possible cause within Tanzu service mesh, the operator would need to use an APM tool like Tanzu Observability to perform more extensive research into the root cause as the problem. APM tools have more capabilities when it comes to providing historical data, performing data analysis and bringing more information such as additional metrics and logs that may not exists in TSM or insights into additional layers that Tanzu service mesh does not touch. The following are the additional views that can be used for troubleshooting:
```
Comment (zmb3) [inline content/outcomes/application-observability/tsm-app-troubleshooting-day2.md:13]: - Say _Tanzu Service Mesh_ (capitalized)
- I'm not sure I would classify TO as an APM tool

File: content/outcomes/application-observability/tsm-app-troubleshooting-day2.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  * Performance tab - in the performance tab we can see performance information on the services and the infrastructure. While the same information is visible in different places, this is a more granular and focused view of this information for better insights. Each subsection can be viewed as Charts which shows the "golden metrics" for each object or the cards which shows an aggregate of all objects depending on how one opts to view the information:
+    * Services - shows each service with its instances and performance information on it: rps, P99 latency and errors rates. Here is the cards view:
```
Comment (zmb3) [inline content/outcomes/application-observability/tsm-app-troubleshooting-day2.md:16]:
```suggestion
    * Services - shows each service with its instances and performance information on it: requests per second (RPS), P99 latency and error rates. Here is the cards view:
```

File: content/outcomes/application-observability/tsm-app-troubleshooting-day2.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    ![Tanzu Service Mesh services cards view](/images/outcomes/app-observability/tsm-services-cards-view.png)  
+    Here is the charts view:
+    ![Tanzu Service Mesh services charts view](/images/outcomes/app-observability/tsm-services-charts-view.png) 
+    * Infrastructure - similarly to the services page in the infrastructure we can see the information on the nodes where the services run in a charts view and a cards view. This covers the nodes' instances count, cpu usage, memory usage, disk usage, and network IO. Here is the cards view:
```
Comment (zmb3) [inline content/outcomes/application-observability/tsm-app-troubleshooting-day2.md:20]:
```suggestion
    * Infrastructure - similar to the services page in the infrastructure view, we can see the information on the nodes where the services run in a charts view and a cards view. This covers the nodes' instances count, CPU usage, memory usage, disk usage, and network IO. Here is the cards view:
```

File: content/outcomes/application-observability/tsm-app-troubleshooting-day2.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    ![Tanzu Service Mesh infrastructure cards view](/images/outcomes/app-observability/tsm-infra-cards-view.png)      
+    The following is the charts view:  
+    ![Tanzu Service Mesh infrastructure charts view](/images/outcomes/app-observability/tsm-infra-charts-view.png)
+  * Cluster view - one of the biggest challenges in distributed applications is correlating between physical and logical objects. Tanzu service mesh addresses that in the cluster view and the node heatmaps. When clicking into the cluster we get a few tabs that allow us to dig a bit deeper into our application health with the following views:
+    * Service topology - similarly to the Global namespaces topology maps, we can see here a dependencies between he services deployed on the cluster, if we click on any of the services we can see the dependencies clearer with the addition of the node heatmap for these specific services. The node heatmap is very useful to correlate between a service and the nodes they are running on while seeing the health of both. In this way we can pinpoint to a overloaded node or if there are any issues coming from the infrastructure.
```
Comment (zmb3) [inline content/outcomes/application-observability/tsm-app-troubleshooting-day2.md:25]:
```suggestion
    * Service topology - similar to the Global namespaces topology maps, we can see here the dependencies between the services deployed on the cluster, if we click on any of the services we can see the dependencies clearer with the addition of the node heatmap for these specific services. The node heatmap is very useful to correlate between a service and the nodes they are running on while seeing the health of both. In this way we can pinpoint to a overloaded node or if there are any issues coming from the infrastructure.  
```

File: content/outcomes/application-observability/tsm-posture-health-perf-distributed-app-day1.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
+---
+date: '2021-07-13'
+lastmod: '2021-07-13'
+layout: single
+team:
+  - Mark Schweighardt
+  - Niran Even-Chen
+  - Andrew Huffman
+title: "Tanzu Service Mesh: Understand the posture (health and performance) of your distributed application (day 1)"
```
Comment (zmb3) [inline content/outcomes/application-observability/tsm-posture-health-perf-distributed-app-day1.md:9]: I would remove `(day 1)` from the title. It's not necessary. If the reader is interested in the outcome of better understanding health and performance of a distributed app, then they will click the link and read the article. Classifying this as day 1 or day 2 doesn't add much.

File: content/outcomes/application-observability/tsm-posture-health-perf-distributed-app-day1.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+weight: 101
+---
+
+In very simple terms, a "modern application" is one that the functions and components are broken into a separate set of micro-services. Each function is considered a service, which is an app on its own, where together they create the "Application". These micro-services communicate with each other over the network and because they can be placed on different infrastructures and clouds, it requires the organization to be able to understand the application posture so that it is easier and faster to figure out what is the problem in case that issues arise. 
+
+The following views in Tanzu Service Mesh help the application owner get insight into how the application is built. Each view can be set in 5 minute, 15 , 30 and 1 hour data sets. This is configured in the top right hand of the screen
```
Comment (zmb3) [inline content/outcomes/application-observability/tsm-posture-health-perf-distributed-app-day1.md:15]: This sentence doesn't read quite right to me, and there's some extra whitespace around the comma separating 15 and 30 minutes.

File: content/outcomes/application-observability/tsm-slo-based-measurement-and-monitoring-apps.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+  - Mark Schweighardt
+  - Niran Even-Chen
+  - Andrew Huffman
+title: "Tanzu Service Mesh: SLO-based Measurement & Monitoring of Applications"
+weight: 103
+---
+
+## Overview
+
+TSM Service level objectives (SLOs) provide a formalized way for Application Operators, Application Owners, and Application Developers to describe, measure, and monitor the performance, quality, and reliability of microservice applications in an automated way.
```
Comment (zmb3) [inline content/outcomes/application-observability/tsm-slo-based-measurement-and-monitoring-apps.md:15]: - Capitalize _Service Level Objectives_
- I might describe what an SLO is in general first, and then say something about how TSM helps. Referring to a "TSM SLO" is a bit confusing.

File: content/outcomes/application-observability/tsm-slo-based-measurement-and-monitoring-apps.md [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+1. **Apply SLO to Services** - You can group services with common characteristics or that are part of one or more related applications in a namespace to form a service group in Tanzu Service Mesh. You can configure and set service groups to automatically track against the same SLO definition. [more](https://docs.vmware.com/en/VMware-Tanzu-Service-Mesh/services/slos-with-tsm/GUID-A13A6444-D48B-40D1-BF52-6151A44D6CEA.html)
+
+1. **View SLO and SLI Violations** - After you define SLO configurations, you can view and track violations of SLIs and SLOs in real time in the Tanzu Service Mesh Console user interface. [more](https://docs.vmware.com/en/VMware-Tanzu-Service-Mesh/services/slos-with-tsm/GUID-F114FF5C-E632-4711-9D5F-B25FD980C5EF.html)
+
+1. **Track “Error Budgets”** - Based on SLOs, teams can track “error budgets,” that is, a length of time over which performance can be outside the desired SLIs without violating the SLO. [more](https://docs.vmware.com/en/VMware-Tanzu-Service-Mesh/services/slos-with-tsm/GUID-9C1D70D7-1C0C-4F72-A39C-42EDCDBB5D9E.html)
```
Comment (zmb3) [inline content/outcomes/application-observability/tsm-slo-based-measurement-and-monitoring-apps.md:42]: We need to better define error budgets here. They aren't necessarily a _length of time_, and there's technically no way to be "outside the desired SLIs" because SLIs are not thresholds.

File: content/outcomes/application-observability/tsm-slo-based-measurement-and-monitoring-apps.md [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+
+1. **Track Autoscaling Impact on Services** - Developers and operators can use the Service Level Objective (SLO) graph to see how Tanzu Service Mesh Service Autoscaler impacts the stated microservice objectives. [more](https://docs.vmware.com/en/VMware-Tanzu-Service-Mesh/services/service-autoscaling-with-tsm-user-guide/GUID-E5684E2C-8D32-4422-BA3B-822606D01E6C.html)
+
+
+## Best Practices
+
+1. **Set High-Level Objectives for Production Environments** – A common measure of the health of a service are latency percentile values. Of the three latency percentiles available to use with Tanzu Service Mesh SLOs, the p99 latency is the most appropriate service level indicator (SLI) for production environments because this metric most fully covers the experience of the users of a service. [more](https://docs.vmware.com/en/VMware-Tanzu-Service-Mesh/services/slos-with-tsm/GUID-2651936B-E8C7-41C3-AE76-5163611970C3.html)
```
Comment (zmb3) [inline content/outcomes/application-observability/tsm-slo-based-measurement-and-monitoring-apps.md:49]: I might say 99th percentile (p99) the first time, so that readers who aren't familiar with the p99 notation know what you're talking about.

File: content/outcomes/application-observability/tsm-app-troubleshooting-day2.md (hunk 2 above)
Comment (ahuffman) [reply content/outcomes/application-observability/tsm-app-troubleshooting-day2.md]: Fixed

File: content/outcomes/application-observability/tsm-app-troubleshooting-day2.md (hunk 3 above)
Comment (ahuffman) [reply content/outcomes/application-observability/tsm-app-troubleshooting-day2.md]: fixed

File: content/outcomes/application-observability/tsm-posture-health-perf-distributed-app-day1.md (hunk 6 above)
Comment (ahuffman) [reply content/outcomes/application-observability/tsm-posture-health-perf-distributed-app-day1.md]: fixed

File: content/outcomes/application-observability/tsm-posture-health-perf-distributed-app-day1.md (hunk 7 above)
Comment (ahuffman) [reply content/outcomes/application-observability/tsm-posture-health-perf-distributed-app-day1.md]: took a crack at fixing this

File: content/outcomes/application-observability/tsm-slo-based-measurement-and-monitoring-apps.md (hunk 8 above)
Comment (ahuffman) [reply content/outcomes/application-observability/tsm-slo-based-measurement-and-monitoring-apps.md]: took a crack at fixing this

File: content/outcomes/application-observability/tsm-slo-based-measurement-and-monitoring-apps.md (hunk 9 above)
Comment (ahuffman) [reply content/outcomes/application-observability/tsm-slo-based-measurement-and-monitoring-apps.md]: took a crack at fixing this

File: content/outcomes/application-observability/tsm-slo-based-measurement-and-monitoring-apps.md (hunk 10 above)
Comment (ahuffman) [reply content/outcomes/application-observability/tsm-slo-based-measurement-and-monitoring-apps.md]: fixed

**Comment by bryanfriedman** [comment]
This PR is pretty outdated at this point. With the latest redesign, don't want to risk merging. If we need to reopen, we can revisit.
